R/vol-surface-class.R

```diff
@@ -0,0 +1,160 @@
+#' VolSurface class
+#'
+#' The `VolSurface`` class is designed to represent volatility surfaces.
+#' Checks whether the dimentions of the surface correspond to the right type of axis: moneyness, delta or strike
+#'
+#' @param reference_date `Date` value that represents the as of date
+#' @param surface_type type of the vol dimentions. Available types: "strike/term"
+#' @param vol_quotes `matrix` of class `numeric` holding the vol values.
+#' @param dim1 `numeric` vector containing the surface strikes, deltas or
+#' moneyness depending on the surface type.
+#' @param dim2 `Date` vector containing the tenors of the surface.
+#' @param ticker label for the vol object, this should be the ticker of the underlying index
+#' @param interpolation object of type interpolation [fmbasics::Interpolation()]
+#' @export
+
+VolSurface <- function(reference_date, surface_type, vol_quotes, dim1, dim2,
```

Comment:
Once you are much clearer on what `vol_quotes` should be, you will be able to simplify this interface considerably.

Reply:
Now the constructor has 4 members, "reference_date", "vol_quotes", "ticker", "interpolation".